samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -445,6 +448,9 @@ private AssessmentGradingData persistAssessmentGrading(ActionEvent ae,
 				log.debug("Submitforgrading: newItemGradingSet.size = {}", adds.size());
 
 				HashSet<ItemGradingData> updateItemGradingSet = getUpdateItemGradingSet(itemGradingSet, adds, calcQuestionMap,imagQuestionMap, emiMap, adata);
+				if ("false".equals(studentRichText)) {
+					updateItemGradingSet.forEach((igd) -> { igd.setAnswerText(ComponentManager.get(FormattedText.class).escapeHtml(igd.getAnswerText(), true)); });
+				}
```

Comment:
what happens if richText is true? is the HTML also escaped?

Reply:
Yes. Now on both cases the info is saved escaped on DB

Reply:
where does it get escaped when studentRichText == true ? Why do we need an if statement? doesn't it need escaping either way?

Reply:
We need that if statement. Everything is fine now.

Reply:
I think antisamy do the work when rich text editor is used @ottenhoff